sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -72,137 +86,129 @@ export class CASSupplierIntegrationService {
-    const activeSupplierAddress = this.getActiveSupplierItemAddress(
-      casSupplierResponseItem,
+    if (!supplierResponse.items.length) {
+      return {
+        status: CASEvaluationStatus.NotFound,
+        reason: NotFoundReason.SupplierNotFound,
+      };
+    }
+    // Check if there is at least one active supplier.
+    const casResponseActiveSupplier = supplierResponse.items.find(
+      (supplier) => supplier.status === "ACTIVE",
     );
-    let supplierAddressToUpdate: SupplierAddress = null;
-    if (activeSupplierAddress) {
-      supplierAddressToUpdate = {
-        supplierSiteCode: activeSupplierAddress.suppliersitecode,
-        addressLine1: activeSupplierAddress.addressline1,
-        addressLine2: activeSupplierAddress.addressline2,
-        city: activeSupplierAddress.city,
-        provinceState: activeSupplierAddress.province,
-        country: activeSupplierAddress.country,
-        postalCode: activeSupplierAddress.postalcode,
-        status: activeSupplierAddress.status,
-        siteProtected: activeSupplierAddress.siteprotected,
-        lastUpdated: new Date(activeSupplierAddress.lastupdated),
+    if (!casResponseActiveSupplier) {
+      return {
+        status: CASEvaluationStatus.NotFound,
+        reason: NotFoundReason.NoActiveSupplierFound,
       };
     }
-    const now = new Date();
-    const systemUser = this.systemUsersService.systemUser;
-    return this.casSupplierRepo.update(
-      {
-        id: casSupplierId,
-      },
-      {
-        supplierNumber: casSupplierResponseItem.suppliernumber,
-        supplierName: casSupplierResponseItem.suppliername,
-        status: casSupplierResponseItem.status,
-        supplierProtected: casSupplierResponseItem.supplierprotected === "Y",
-        lastUpdated: new Date(casSupplierResponseItem.lastupdated),
-        supplierAddress: supplierAddressToUpdate,
-        supplierStatus,
-        supplierStatusUpdatedOn: now,
-        isValid: true,
-        updatedAt: now,
-        modifier: systemUser,
-      },
+    // Get a matching address, if exists.
+    // Address must be active and have the same address line 1
+    // and postal code considering the CAS formats.
+    const casFormattedStudentAddress = formatAddress(
+      casSupplier.student.contactInfo.address.addressLine1,
+    );
+    const casFormattedPostalCode = formatPostalCode(
+      casSupplier.student.contactInfo.address.postalCode,
     );
+    const casResponseMatchedAddress =
+      casResponseActiveSupplier.supplieraddress?.find((address) => {
+        return (
+          address.status === "ACTIVE" &&
+          address.addressline1 === casFormattedStudentAddress &&
+          address.postalcode === casFormattedPostalCode
+        );
+      });
+    return {
+      status: CASEvaluationStatus.ActiveSupplierFound,
+      activeSupplier: casResponseActiveSupplier,
+      matchedAddress: casResponseMatchedAddress,
```

Comment:
That was the initial approach which would require one more processor.
I ended up reverting it because these two pieces of information are tightly coupled.
I will give it a second try in preparation for the upcoming address site creation.
- Active supplier and site found: `update supplier and site`.
- Active supplier found with no address match: create the site, and `update supplier and site`.
The `create the site` will happen inside a new processor that will call a shared method to then `update supplier and site`.
Does it sound ok?

Reply:
Created a new status and processor. The method for `update supplier and site` was kept inside each processor.